src/resolve/resolvectl.c

```diff
@@ -476,7 +476,7 @@ static int resolve_record(sd_bus *bus, const char *name, uint16_t class, uint16_
 
         log_debug("Resolving %s %s %s (interface %s).", name, dns_class_to_string(class), dns_type_to_string(type), isempty(arg_ifname) ? "*" : arg_ifname);
 
-        if (single_label_nonsynthetic(name))
+        if (!dns_name_dot_suffixed(name) && single_label_nonsynthetic(name))
```

Comment:
this returns "int", not "bool", i.e. can fail. i.e. please acknowledge that in code, and do a <= 0 or == 0 or so

Reply:
Oh, indeed. Not sure how I missed that.